backend/database/migrations/2021_01_27_213544_insert_user_roles.php

```diff
@@ -40,8 +27,11 @@ public function up()
      */
     public function down()
     {
-        Schema::disableForeignKeyConstraints();
-        DB::table('roles')->truncate();
-        Schema::enableForeignKeyConstraints();
+        $roles = Role::getArrayOfAllRoleNames();
+        foreach ($roles as $role) {
+            /** @var $role Role */
+            $role = Role::findByName($role);
+            $role->delete();
+        }
```

Comment:
Thanks for this.  I like this a lot better than relying on truncate to do the job.